lua-mode.el

```diff
@@ -86,12 +86,11 @@
 
 ;;; Code:
 (eval-when-compile
-  (require 'cl))
-
+  (require 'cl)
+  (require 'subr-x))
```

Comment:
We should probably add `(emacs "24.4")` to package dependencies in `Cask` (https://github.com/purcell/package-lint/issues/110 for reference)

Reply:
Right.  Again I'm not a Cask expert; would you mind adding the correct incantations and submitting to to this PR branch?  I'll stop rewriting pushed history until we converge.  Anything that needs to be placed elsewhere just add a `[TOSQUASH]`.